packages/puppeteer-core/src/cdp/Target.ts

```diff
@@ -65,16 +65,16 @@ export class CdpTarget extends Target {
     this.#browserContext = browserContext;
     this._targetId = targetInfo.targetId;
     this.#sessionFactory = sessionFactory;
-    if (this.#session && this.#session instanceof CdpCDPSession) {
+    if (this.#session) {
       this.#session.setTarget(this);
     }
   }
 
   override async asPage(): Promise<Page> {
     const session = this._session();
     if (!session) {
-      return await this.createCDPSession().then(client => {
-        return CdpPage._create(client, this, null);
+      return await this._createCDPSession().then(session => {
```

Comment:
Don't rename the handle in the promise it's consing to have to variable in the same scope with the same name, But its a pattern we call Session clients as well.